src/itype.h

```diff
@@ -607,11 +607,18 @@ struct itype {
 
 // Includes food drink and drugs
 struct it_comest : itype {
+    friend class Item_factory;
```

Comment:
It's best to stick with a constructor and make the object immutable. If we
want to selectively override fields it's better to replace the object with
a new instance with the new values.
Setters would suggest to people that the fields can be modified at runtime
(which has happened in the past and caused hard to find bugs).